### PR TITLE
CMake: corrections and updates

### PR DESCRIPTION
**Pull request by scivision**
I helped implement CMake for stdlib, and some syntax and techniques have become obsolete or no longer relevant. At the time I implemented, quite old compilers were being considered, which have long since not been supported by stdlib. 

* remove discontinued and never worked with stdlib PGI
* remove syntax for unsupported old Intel compiler versions
* correct Intel compiler options for Windows

### CODE REVIEW

**Comment by scivision** [comment]
Note: it appears there is a tricky oneAPI MacOS bug, that bypasses my ability to detect as it doesn't show up until building whole stdlib. 

Can we get the 2021.3 oneAPI which fixed numerous other bugs, perhaps it fixes this too. I don't know the URL to download it to the CI.

**Review by jvdp1 — APPROVED** [review]
Thank you. LGTM.

**Review by awvwgk — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,21 +31,20 @@ if(CMAKE_Fortran_COMPILER_ID STREQUAL GNU)
...
-elseif(CMAKE_Fortran_COMPILER_ID STREQUAL PGI)
-  add_compile_options(-Mdclchk)
 endif()
 
 # --- compiler feature checks
 include(CheckFortranSourceCompiles)
 include(CheckFortranSourceRuns)
-check_fortran_source_runs("i=0; error stop i; end" f18errorstop SRC_EXT f90)
+check_fortran_source_runs("i=0; error stop i; end" f18errorstop)
 check_fortran_source_compiles("real, allocatable :: array(:, :, :, :, :, :, :, :, :, :); end" f03rank SRC_EXT f90)
```
Comment (ivan-pi) [inline CMakeLists.txt:48]: Is their any reason to keep the source extension in the rank test on (new) line 48?

**Review by ivan-pi — APPROVED** [review]
I left a comment, but otherwise +1 to merge.

File: CMakeLists.txt (hunk 1 above)
Comment (scivision) [reply CMakeLists.txt:48]: check_fortran_source_compiles is an older command that requires SRC_EXT. I wrote check_fortran_source_runs for CMake 3.14 without that requirement.

**Comment by jvdp1** [comment]
All questions seem to be answered. I'll merge it. Thank you @scivision for this PR.

**Pull request merged**